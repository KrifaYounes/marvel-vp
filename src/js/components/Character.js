import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Col } from 'react-flexbox-grid';
import { Link } from 'react-router-dom';

class Character extends PureComponent {

    render() {
        const character = this.props.character;
        const imgSource = character.thumbnail.path + '.' + character.thumbnail.extension;
        const urls = this.props.character.urls;

        const urlInfos = urls.map( (url, index) => {
            return (<a key={index} href={url.url}> {url.type} </a>)
        });

        return (
            <Col className='character'>
                <Link to={'/' + character.id } className='character-name'>
                    <div><img src={imgSource}  alt='MARVEL'/></div>
                    <div>{character.name}</div>
                </Link>

                <div className='urlInfos'>
                    { urlInfos }
                </div>
            </Col>
        );
    }
}


Character.propTypes = {
    character: PropTypes.object.isRequired
};

export default Character;