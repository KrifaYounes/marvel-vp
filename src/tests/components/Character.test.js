import React from 'react';
import Character from '../../js/components/Character';
import { shallow } from 'enzyme';
import enzymeToJSON from 'enzyme-to-json';

test('Character Component Snapshot', () => {
    const character = {
        urls : [],
        thumbnail : {
            path : 'http://marvel.com/image',
            extension: 'jpeg'
        }
    };

    const wrapper = shallow(<Character character={character}/>);
    //must match the snapshot

    expect(enzymeToJSON(wrapper)).toMatchSnapshot();
});
