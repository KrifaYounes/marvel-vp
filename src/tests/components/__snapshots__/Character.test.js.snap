// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Component Snapshot 1`] = `
<Col
  className="character"
>
  <Link
    className="character-name"
    replace={false}
    to="/undefined"
  >
    <div>
      <img
        alt="MARVEL"
        src="http://marvel.com/image.jpeg"
      />
    </div>
    <div />
  </Link>
  <div
    className="urlInfos"
  />
</Col>
`;
